src/scanspec/core.py

```diff
@@ -616,7 +616,7 @@ def consume(self, num: int | None = None) -> Frames[Axis]:
 
     def __len__(self) -> int:
         """Number of frames left in a scan, reduces when `consume` is called."""
-        return self.end_index - self.index
+        return int(self.end_index - self.index)
```

Comment:
Nit: if we define `self.end_index: int = np.prod...` then we shouldn't have to do this?

Reply:
That still requires a cast when I try it...

Reply:
In the end I went with `self.end_index = int(np.prod(self.lengths))`, since that fixes another error that emerged with a pyright upgrade